File: testapp/python-backend/services/food_service.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', '..'))

from testapp.generated.interfaces.i_food_service import IfoodService


class FoodService(IfoodService):
    """YOUR FILE — edit freely, Veld never overwrites it."""

    def __init__(self):
        self._foods = [
            {'id': '1', 'name': 'Pizza',  'price': 12, 'tags': ['italian', 'hot'], 'type': 'meat'},
            {'id': '2', 'name': 'Burger', 'price': 8,  'tags': ['american', 'fast-food'], 'type': 'burger'},
            {'id': '3', 'name': 'Salad',  'price': 7,  'tags': ['healthy', 'cold'], 'type': 'salatat'},
        ]

    def get_all_foods(self, user_id):
        return {'items': self._foods, 'total': len(self._foods)}

    def add_food(self, input):
        food = {'id': str(len(self._foods) + 1), **input}
        self._foods.append(food)
        return food

File: testapp/backend/src/services/FoodService.ts
import type { Food, FoodList, CreateFoodInput } from '../../../generated/types/food'
import type { IfoodService } from '../../../generated/interfaces/IfoodService'

export class FoodService implements IfoodService {

  private foods: Food[] = [
    { id: '1', name: 'Pizza',  price: 12, tags: ['italian', 'hot'] },
    { id: '2', name: 'Burger', price: 8,  tags: ['american', 'fast-food'] },
    { id: '3', name: 'Salad',  price: 7,  tags: ['healthy', 'cold'] },
  ]

  async GetAllFoods(): Promise<FoodList> {
    return { items: this.foods, total: this.foods.length }
  }

  async AddFood(input: CreateFoodInput): Promise<Food> {
    const food: Food = { id: String(this.foods.length + 1), ...input }
    this.foods.push(food)
    return food
  }
}
